cli/ballerina-cli-module/src/main/java/org/ballerinalang/cli/module/Pull.java

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.ballerinalang.cli.module;
+
+import me.tongfei.progressbar.ProgressBar;
+import me.tongfei.progressbar.ProgressBarStyle;
+import org.ballerinalang.cli.module.util.ErrorUtil;
+import org.ballerinalang.cli.module.util.Utils;
+import org.ballerinalang.jvm.JSONParser;
+import org.ballerinalang.jvm.values.MapValue;
+
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.ballerinalang.cli.module.util.Utils.convertToUrl;
+import static org.ballerinalang.cli.module.util.Utils.createHttpUrlConnection;
+import static org.ballerinalang.cli.module.util.Utils.initializeSsl;
+import static org.ballerinalang.cli.module.util.Utils.setRequestMethod;
+
+/**
+ * This class is pulling the modules from ballerina central.
+ */
+public class Pull {
+    private static final String VERSION_REGEX = "(\\d+\\.)(\\d+\\.)(\\d+)";
+    private static PrintStream outStream = System.out;
+    private static DefaultLogFormatter logFormatter = new DefaultLogFormatter();
+
+    private Pull() {
+    }
+
+    public static void execute(String url, String modulePathInBaloCache, String modulePath, String proxyHost,
+            int proxyPort, String proxyUsername, String proxyPassword, String supportedVersionRange, boolean isBuild,
+            String nightlyBuild, String langSpecVersion, String platform) {
+        if (isBuild) {
+            logFormatter = new BuildLogFormatter();
+        }
+
+        initializeSsl();
+        HttpURLConnection conn = createHttpUrlConnection(convertToUrl(url + supportedVersionRange), proxyHost,
+                proxyPort, proxyUsername, proxyPassword);
+        conn.setInstanceFollowRedirects(false);
+        setRequestMethod(conn, Utils.RequestMethod.GET);
+
+        // set headers.
+        if (!platform.equals("")) {
```

Comment:
Can `platform` become `null`? Should we use null safe check? i.e. `"".equals(platform)`?